frontend/src/app/main/ui/workspace/snap_distances.cljs

```diff
@@ -185,19 +185,24 @@
         lt-distances (->> lt-shapes (map distance-to-selrect) (filter pos?) (into #{}))
         gt-distances (->> gt-shapes (map distance-to-selrect) (filter pos?) (into #{}))
 
-
         ;; We'll show the distances that match a distance from the selrect
         show-candidate? #(check-in-set % (set/union lt-distances gt-distances))
 
         ;; Checks the distances between elements for distances that match the set of distances
         distance-coincidences (concat (get-shapes-match show-candidate? lt-shapes)
                                       (get-shapes-match show-candidate? gt-shapes))
 
+
         ;; Show the distances that either match one of the distances from the selrect
         ;; or are from the selrect and go to a shape on the left and to the right
-        show-distance? #(check-in-set % (into #{} (concat
-                                                   (map first distance-coincidences)
-                                                   (set/intersection lt-distances gt-distances))))
+        show-distance?
+        (fn [dist]
+          (let [distances-to-show
+                (->> (d/concat (mapv first distance-coincidences)
```

Comment:
You can done all this in one step just using `#{}` as a first value of d/concat. 
Also, the use of `filterv` should be replace `filter` instead I think (because you are converting seq to vec and later to seq (inside d/concat).

Reply:
OK